### PR TITLE
dim: init at 0-unstable-2023-12-29

### LINKED ISSUE #142736: Packaging Request: Dim
**Project description**

Dim is a self-hosted media manager. With minimal setup, Dim will organize and beautify your media collections, letting you access and play them anytime from anywhere.

**Metadata**

* homepage URL: https://github.com/Dusk-Labs/dim
* source URL: https://github.com/Dusk-Labs/dim
* license: GNU General Public License v2.0
* platforms: Unix

### PR DESCRIPTION
**Pull request by Misterio77**
## Description of changes

Adds [dim](https://github.com/Dusk-Labs/dim), a self-hosted media manager.

Fixes #142736

## Things done

Dim uses a hacky approach to [find `ffmpeg`/`ffprobe`](https://github.com/Dusk-Labs/dim/blob/3ccb4ab05fc1d7dbd4ebbba9ff2de0ecc9139b27/dim-core/src/streaming/mod.rs#L12-L24) and its `config`/`log` dir: it only works if they are in [the same directory as the binary](https://github.com/Dusk-Labs/dim/blob/3ccb4ab05fc1d7dbd4ebbba9ff2de0ecc9139b27/dim-utils/src/lib.rs#L403-L410), which is a no-go for us.

To fix that, I patched it to use `ffmpeg`/`ffprobe` from PATH as usual, and access `config`/`log` relative to the working directory. This works for now, but I'd like to see if I can come up with a solution that we can upstream. Perhaps make them configurable through env vars/CLI args.

I plan on also making a NixOS module in the next couple of weeks.

Oh, and [it currently requires ffmpeg 5](https://github.com/Dusk-Labs/dim/issues/547).

---

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/di/dim/package.nix [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  ffmpeg_5,
+  git,
+  pkg-config,
+  sqlite,
+  libvaSupport ? stdenv.hostPlatform.isLinux,
+  libva,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "dim";
+  version = "unstable-2023-12-29";
```
Comment (AndersonTorres) [inline pkgs/by-name/di/dim/package.nix:17]:
```suggestion
  version = "0-unstable-2023-12-29";
```

new rules: version should start with a digit.

File: pkgs/by-name/di/dim/package.nix [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  src = fetchFromGitHub {
+    owner = "Dusk-Labs";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/by-name/di/dim/package.nix:20]:
```suggestion
    repo = "dim";
```

https://github.com/NixOS/nixpkgs/issues/277994

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by rodrgz** [comment]
Result of `nixpkgs-review pr 289318` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dim</li>
  </ul>
</details>

**Review by rodrgz — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

File: pkgs/by-name/di/dim/package.nix [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    hash = "sha256-1mgbrDnIkIdWy78uj4EjjgwBQxw/rIS1LCFNscXXPbk=";
+  };
+
+  frontend = buildNpmPackage {
+    pname = "dim-ui";
+    inherit version;
+    src = "${src}/ui";
+
+    postPatch = ''
+      ln -s ${./package-lock.json} package-lock.json
```
Comment (dotlambda) [inline pkgs/by-name/di/dim/package.nix:31]: Can we ask upstream to update their `yarn.lock` and use `fetchYarnDeps` instead?

Comment (lucasew) [reply pkgs/by-name/di/dim/package.nix:31]: Or update it on the postFetch if they don't look much collaborative.

https://github.com/NixOS/nixpkgs/pull/261139

**Pull request merged**